modules/http/src/main/java/org/mule/module/http/internal/listener/grizzly/ResponseStreamingCompletionHandler.java

```diff
@@ -66,28 +66,37 @@ public void sendInputStreamChunk() throws IOException
     {
         final Buffer buffer = memoryManager.allocate(DEFAULT_BUFFER_SIZE);
 
-        final byte[] bufferByteArray = buffer.array();
         final int offset = buffer.arrayOffset();
         final int length = buffer.remaining();
 
-        int bytesRead = inputStream.read(bufferByteArray, offset, length);
-        final HttpContent content;
-
-        if (bytesRead == -1)
-        {
-            content = httpResponsePacket.httpTrailerBuilder().build();
-            isDone = true;
-        }
-        else
-        {
-            buffer.limit(bytesRead);
-            content = httpResponsePacket.httpContentBuilder().content(buffer).build();
-        }
+        isDone = readStreamManually(buffer);
 
+        HttpContent content = httpResponsePacket.httpContentBuilder().content(buffer).build();
         ctx.getConnection().getAttributes().setAttribute(MULE_CLASSLOADER, loggerClassLoader);
         ctx.write(content, this);
+
+        if (isDone) {
+            content = httpResponsePacket.httpContentBuilder().build();
```

Comment:
You should use `httpTrailerBuilder` as it was before.
Even though the last chunk in a transfer encoding is represented by an empty chunk, we don't know whether Grizzly's internals do some special logic depending on te type of the `HttpContent` instance.